eng/testing/runsettings.targets

```diff
@@ -0,0 +1,40 @@
+<Project InitialTargets="GenerateRunSettingsFile">
+
+  <PropertyGroup>
+    <RunSettingsInputFilePath>$(MSBuildThisFileDirectory).runsettings</RunSettingsInputFilePath>
+    <RunSettingsAppOutputFilePath>$(OutDir).runsettings</RunSettingsAppOutputFilePath>
+
+    <RunSettingsOutputFilePath>$(RunSettingsAppOutputFilePath)</RunSettingsOutputFilePath>
+
+    <ResultsFileName>$(MSBuildProjectName)_$(TargetFramework)_$(TargetPlatform)</ResultsFileName>
+
+    <RunSettingsFilePath Condition="Exists('$(RunSettingsOutputFilePath)')">$(RunSettingsOutputFilePath)</RunSettingsFilePath>
+    <VSTestRunSettingsFile Condition="'$(RunSettingsFilePath)' != ''">$(RunSettingsFilePath)</VSTestRunSettingsFile>
+  </PropertyGroup>
+
+  <Target Name="GenerateRunSettingsFile"
+          Inputs="$(RunSettingsInputFilePath)"
+          Outputs="$(RunSettingsOutputFilePath)">
+
+    <PropertyGroup>
+      <RunSettingsFileContent>$([System.IO.File]::ReadAllText('$(RunSettingsInputFilePath)'))</RunSettingsFileContent>
+      <RunSettingsFileContent>$(RunSettingsFileContent.Replace('$$RESULTSDIRECTORY$$', '$(ArtifactsTestResultsDir)')
+                                                      .Replace('$$SOLUTIONDIRECTORY$$', '$(RepoRoot)')
+                                                      .Replace('$$TARGETPLATFORM$$', '$(TargetPlatform)')
+                                                      .Replace('$$TRXRESULTSNAME$$', '$(ResultsFileName).xml')
```

Comment:
TRX files should have the `.trx` extension.

Reply:
Arcade sets `.xml` extension to them. I'm doing this to get the same file extension. I'll open an issue in arcade.

Reply:
Yes, see our discussion here: https://github.com/dotnet/arcade/pull/6355#issuecomment-708305947.

Reply:
I opened: https://github.com/dotnet/arcade/issues/6419